### PR TITLE
docs: add design doc for rule based index selection

### LINKED ISSUE #27227: Create proposal of heuristic rules enhancement for index selection

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #27227

### Release note

```release-note
None
```

### CODE REVIEW

File: docs/design/2021-07-07-rule-based-index-selection.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Proposal: Improve rule based index selection
+
+- Author(s): [xuyifangreeneyes](https://github.com/xuyifangreeneyes)
+- Last updated: 2021-08-13 
+- Discussion at:
```
Comment (tisonkun) [inline docs/design/2021-07-07-rule-based-index-selection.md:5]:
```suggestion
- Author(s): [xuyifangreeneyes](https://github.com/xuyifangreeneyes)
- Discussion PR: https://github.com/pingcap/tidb/pull/27223
- Tracking Issue: https://github.com/pingcap/tidb/issues/26020
```

Please follow the template https://raw.githubusercontent.com/pingcap/tidb/master/docs/design/TEMPLATE.md

**Comment by tisonkun** [comment]
/cc @bb7133 @yudongusa 

I think this proposal may require a release note as user should be aware that we improve performance in several scenarios.

**Comment by winoros** [comment]
@tisonkun Release note has contained this one.

**Review by winoros — APPROVED** [review]

**Comment by tisonkun** [comment]
/uncc @bb7133 @yudongusa 

/cc @rebelice @qw4990 

Please take a look at the late design document. Thanks!

**Review by qw4990 — APPROVED** [review]

**Comment by tisonkun** [comment]
@winoros @qw4990 could you please help on merging this PR?

**Pull request merged**